### PR TITLE
[CGal plugin][CImgPlugin] Image data moved from Image/ to CImgPlugin/

### PR DESCRIPTION
**Pull request by omarboukhris**
This "move" is originally done to uncouple CGAL plugin and Image. This was done by moving Image class (along with dependencies) from Image plugin (ImageTypes.h) to CImgPlugin (CImgData.h).

The point is to reduce compilation time since CImg takes less time to compile than Image.
Now Image and CGAL both depend on CImgPlugin which is lighter.
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: applications/plugins/CGALPlugin/CGALPlugin_test/MeshGenerationFromImage_test.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@
 #define CGALPLUGIN_MESHGENERATIONFROMIMAGE_CPP
 
 #include <CGALPlugin/MeshGenerationFromImage.h>
-#include <image/ImageTypes.h>
+//#include <image/ImageTypes.h>
```
Comment (epernod) [inline applications/plugins/CGALPlugin/CGALPlugin_test/MeshGenerationFromImage_test.cpp]: you can remove the line definitively

File: applications/plugins/CGALPlugin/MeshGenerationFromImage.h [hunk 2]
```diff
@@ -45,7 +45,9 @@
 #include <sofa/defaulttype/Mat.h>
 #include <sofa/defaulttype/Quat.h>
 #include <sofa/helper/rmath.h>
-#include <image/ImageTypes.h>
+
+//#include <image/ImageTypes.h>
```
Comment (epernod) [inline applications/plugins/CGALPlugin/MeshGenerationFromImage.h]: same here. You can remove the line.

File: applications/plugins/CImgPlugin/CMakeLists.txt [hunk 3]
```diff
@@ -86,7 +87,7 @@ add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES})
 
 target_compile_options(${PROJECT_NAME} PUBLIC ${CIMG_CFLAGS})
 target_compile_options(${PROJECT_NAME} PRIVATE "-DSOFA_BUILD_CIMGPLUGIN")
-target_link_libraries(${PROJECT_NAME} SofaCore ${EXTERNAL_LIBS})
+target_link_libraries(${PROJECT_NAME} SofaCore SofaSimulationCommon ${EXTERNAL_LIBS})
```
Comment (epernod) [inline applications/plugins/CImgPlugin/CMakeLists.txt]: SofaSimulationCommon  is needed?

**Review by epernod — COMMENTED** [review]
it is a shame that git doesn't see the move from ImageTypes to ImageData as the same file to see only the real modifications.
I suppose you didn't change the file except the includes and class names ?

File: applications/plugins/CImgPlugin/CMakeLists.txt (hunk 3 above)
Comment (omarboukhris) [reply applications/plugins/CImgPlugin/CMakeLists.txt]: it is not, forgot how it ended there.
I removed it, along with the commented includes and pushed the updated code to remote.

Comment (omarboukhris) [reply applications/plugins/CImgPlugin/CMakeLists.txt]: I didn't change the files/code/algos, I just moved classes around.

**Review by epernod — APPROVED** [review]

**Comment by guparan** [comment]
[ci-build][with-all-tests]

**Pull request merged**